### DISCUSSION TITLE
Support custom tags

### DISCUSSION THREAD

**Discussion by kettanaito** [ossjs/release#19] [Ideas]
It'd be nice to give an option to publish a certain version under a custom tag (i.e. "beta"). Right now, all versions are published with the "latest" tag.

**Comment by kettanaito** [comment]
## Proposal

### Usage

```sh
release publish --tag beta
```

### Release version

Running `pubilsh` with the custom `tag` value will still calculate the next release version based on the commits you have, and create the next release version in this format:

- Given previous published version: `1.2.3`.
- And the following commit: `feat: new stuff`.
- It'd use the next release version: `1.3.0-beta`.

**Reply by jbcpollak** [reply]
If you go this route, I would suggest `--prerelease` as the flag because that's what semver calls it.

**Comment by jbcpollak** [comment]
A beta release might have the following change made to the config file:

```
module.exports = {
    script: "npm release",
    prerelease: "beta"
};
```

In this case, semrel is calculated as normal assuming it is up-to-date with all versions already released. 

Releases here use regular semrel rules, but add an incrementing `-beta.X` if a matching semrel tag already exists. 

For example: `0.1.1` -> `0.1.2-beta.1` -> `0.1.2-beta.2`, but if you then commit a feature, the next release will be `0.2.0-beta-1`.

I kind of prefer the use of a config setting over a CICD pipeline change, because those can often be hard to hunt down and might have additional process around changing them.

**Reply by kettanaito** [reply]
I think "prerelease" (as in "treat breaking changes as minor") and custom tags like "beta" are two separate things, which should also be configured separately. 

For example.

```js
module.exports = {
  script: '...',
  releaseType: 'prerelease'
}
```

This releases breaking as `x.Y.x` but uses `latest` as the release tag.

Adding a custom `tag` option/configuration property could complement this behavior, or be a standalone thing. 

```sh
release publish --tag beta
```

My reasoning for putting `tag` into the inline option is that you never want to release your software as `beta` as a default tag. The default should still be `latest` because that's what NPM pulls in. Nothing stops you from having two `ossjs.release` configs, I suppose, where one would have `tag: 'beta'` hard-coded, and then use `release publish -c ossjs.release.beta.config.js` or something. 

Not sure to which extent I want to make this customizable. `--tag` is an escape hatch to explicitly publish a version under a tag. I'd expect this to be called _manually_, and to be a conscious choice.

**Comment by kettanaito** [comment]
This is already possible via `profile[n].name` and explicit `--tag` in your publish command:

```js
// release.config.json
{
  "$schema": "./node_modules/@ossjs/release/schema.json",
  "profiles": [
    {
      "name": "latest",
      "use": "npm publish"
    },
    {
      "name": "nightly",
      "use": "npm publish --tag nightly"
    }
  ]
}
```

There's no smart logic between the `name` and your command. You have to keep it in-sync yourself, which isn't hard to do given tags do not change. Create as many profiles as you need, e.g. for the latest tag, backport, beta, next, nightly, etc.
